### PR TITLE
Codegen: allow `synth` properties of non-`synth` classes

### PR DESCRIPTION
**Pull request by redsun82**
This can serve as a basis for https://github.com/github/codeql/pull/13240.

### CODE REVIEW

File: misc/codegen/test/test_qlgen.py [hunk 1]
```diff
@@ -861,25 +861,47 @@ def test_property_on_class_with_default_doc_name(generate_classes):
...
+                            tableparams=["this", "result"], doc="foo of this my object"),
+            ])),
     }
 
 
 def test_stub_on_class_with_ipa_on_arguments(generate_classes):
     assert generate_classes([
-        schema.Class("MyObject", ipa=schema.IpaInfo(on_arguments={"base": "A", "index": "int", "label": "string"})),
+        schema.Class("MyObject", ipa=schema.IpaInfo(on_arguments={"base": "A", "index": "int", "label": "string"}),
+                     properties=[schema.SingleProperty("foo", "bar")]),
```
Comment (d10c) [inline misc/codegen/test/test_qlgen.py:880]: Am I reading this correctly that when you add a simple `foo: bar` property, it becomes `synth=True` by default? Or is it because this class is declared as inheriting from class "A" and its `foo: bar | synth` property on line 575 of test_dbschemegen.py? Basically, I'm finding it hard to follow how the fixtures for this test are set up.

**Review by d10c — COMMENTED** [review]
Looks good! Left a minor comment. Do you think DCA is warranted for this PR?

File: misc/codegen/test/test_qlgen.py (hunk 1 above)
Comment (d10c) [reply misc/codegen/test/test_qlgen.py:880]: Oh, right, I didn't realize that that's what `ipa=` was doing. Doing that renaming would indeed be clearer, yes. I thought that this PR did the complete renaming, but apparently it's backwards-compatible with `ipa`, so it's probably also fine as it is.

**Comment by d10c** [comment]
Hi @redsun82, can we resolve the conflicts and merge this soon-ish? I intend to build https://github.com/github/codeql/pull/13240 on top of this PR, which would obviate the need for a dbscheme change.

**Comment by redsun82** [comment]
sorry about the delay @d10c! I've done the renaming (which was just a matter of `s/Ipa/Synth/g` and `s/ipa/synth/g`, plus the renaming of two templates). Thank you for the merge and the fix commits! This can be merged now as far as I'm concerned.

**Review by d10c — APPROVED** [review]
LGTM! (Though I think someone else also has to approve?)

**Comment by redsun82** [comment]
I think it's ok to merge it with your approval. This PR in itself introduces no changes in actually generated code, so it's 0 risk. If when using this feature you'll find issues, we can come back to it with follow up changes.
